spring-boot-admin/Dockerfile

```diff
@@ -1,21 +1,40 @@
 # -----------------------------------------------------------------------------
-# Spring Boot Admin Server build
+# Spring Boot Admin Server Build
 # -----------------------------------------------------------------------------
 
-FROM bellsoft/liberica-openjdk-alpine:8u252 as build
-ENV IMAGE_VERSION 2.2.3
-ENV GRADLE_OPTS -Dorg.gradle.daemon=false
+FROM eclipse-temurin:21-alpine AS build
 WORKDIR /scratch
-COPY files .
-RUN ./gradlew build
-RUN ln -s build/libs/springbootadmin-${IMAGE_VERSION}.jar springbootadmin.jar
+RUN apk update && apk add ca-certificates && apk add curl && apk add patch
+RUN curl --get https://start.spring.io/starter.zip \
+        -d "type=gradle-project" \
+        -d "platformVersion=3.3.4" \
+        -d "jvmVersion=21" \
+        -d "groupId=io.steeltoe.docker" \
+        -d "artifactId=springbootadmin" \
+        -d "applicationName=SpringBootAdmin" \
+        -d "language=java" \
+        -d "dependencies=codecentric-spring-boot-admin-server" \
+        --output springbootadmin.zip
+RUN mkdir springbootadmin && unzip -d springbootadmin springbootadmin.zip
+COPY metadata metadata
+COPY patches patches
+RUN sed -i "s/<VERSION>/$(cat metadata/IMAGE_VERSION)/g" patches/version.patch
+RUN for patch in patches/*.patch; do \
+        echo "applying patch $(basename $patch)"; \
+        cd springbootadmin; \
+        patch -p1 < ../$patch; \
+        cd ..; \
+        done
+RUN springbootadmin/gradlew bootJar --project-dir springbootadmin --no-daemon
+RUN mkdir output && \
+    cp "springbootadmin/build/libs/springbootadmin-$(cat metadata/IMAGE_VERSION).jar" output/springbootadmin.jar
 
 # -----------------------------------------------------------------------------
-# Spring Boot Admin Server image
+# Spring Boot Admin Server Linux Image
 # -----------------------------------------------------------------------------
 
-FROM bellsoft/liberica-openjre-alpine:8u252
-WORKDIR /spring-boot-admin
-COPY --from=build /scratch .
-EXPOSE 8080
+FROM eclipse-temurin:21-alpine
```

Comment:
The config server dockerfile uses "FROM eclipse-temurin:21" (without alpine suffix), is that intentional?

Reply:
While it wasn't changed in this PR, I see no reason not to use alpine, so I've changed it to match the others.